### DISCUSSION TITLE
Consumption Cost

### DISCUSSION THREAD

**Discussion by d1964b** [BottlecapDave/HomeAssistant-OctopusEnergy#297] [Q&A]
It's been over a day now since my 'octopus_energy:electricity_{{METER_SERIAL_NUMBER}}_{{MPAN_NUMBER}}_previous_accumulative_cost' value last changed , I'm just wondering what the trigger is ? Thanks

**Comment by BottlecapDave** [comment]
The previous consumption usually takes up to 24 hours to refresh, but is at the mercy of OE getting the data from your meter in due time. The sensor only ever looks at the last 24 hours, so if it takes longer for OE to get the data you won't get it in HA :(

Do you have any errors in your logs which might indicate another error?

**Comment by d1964b** [comment]
The value has come through now ( at 19:10 ) which is about 32 hrs after the previous update and a little over half an hour after restarting HA. If I look in the Device settings of the integration it says ‘No Logbook Entries Found’ and since I’m not sure if that is the best place to check I also looked in the Core Logs and there was this , but I don’t see anything obvious to suggest they are related.

2023-07-04 18:30:00.015 ERROR (MainThread) [homeassistant] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
result = self.fn(*self.args, **self.kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

I logged in to my Octopus account at about midday and it had yesterday’s readings at that point. It’s not the end of the world or anything , I was just curious about the trigger in case there was a manual way of kicking of the data retrieval or something.

Thanks for your work.
